Tensile/Source/lib/include/Tensile/MasterSolutionLibrary.hpp

```diff
@@ -134,15 +135,21 @@ namespace Tensile
 
                     if((*selected_solution->problemPredicate)(problem)
                        && (*selected_solution->hardwarePredicate)(hardware))
-                        return selected_solution;
+                        rv = selected_solution;
                     else
                         return nullptr;
                 }
             }
             else
-                return library->findBestSolution(problem, hardware, fitness);
-        }
+                rv = library->findBestSolution(problem, hardware, fitness);
 
+            if(Debug::Instance().printLibraryLogicIndex() && rv)
+            {
+                std::cout << "Library logic solution index of winning solution: "
```

Comment:
nit: I'd vote to have a separate message for no solution too, more info is probably better if we're debugging

Reply:
Sure. The user would probably find out pretty quick anyways if no solution is found, but having a message for logging purposes could be nice. How's a simple "No solution found" sound?